### PR TITLE
multi: add some route blinding helpers

### PR DESCRIPTION
**Pull request by ellemouton**
1) let `BuildBlindedPath` return the very last ephemeral public key. This is how a path builder (the recipient) can identify an incoming payment as belonging to a specific payment they generated earlier. 
2) add `NextEphemeralPriv`, the private key equivalent of `NextEphemeral`

### CODE REVIEW

File: sphinx.go [hunk 1]
```diff
@@ -187,7 +187,7 @@ func generateSharedSecrets(paymentPath []*btcec.PublicKey,
 		)
 	}
 
-	return hopSharedSecrets, nil
+	return hopSharedSecrets, lastEphemeralPubKey, nil
```
Comment (Roasbeef) [inline sphinx.go:190]: Perhaps time to make into a struct return value? 

Would any of the other intermediate values be useful here? Maybe for doing self-processing of non dummy hops in a blinded path?

Comment (ellemouton) [reply sphinx.go:190]: the only other thing we need to do the self-processing is the session key which we anyways pass in to the package from the above layer. 

But yeah - happy to turn this into a struct if better!

Comment (ellemouton) [reply sphinx.go:190]: updated the exported `BuildBlindedPath` function to return a new `BlindedPathInfo` struct

**Review by Roasbeef — APPROVED** [review]
LGTM 🧉

**Pull request merged**